itests/qtest-iceberg/src/test/resources/gravitino-h2-test-template.conf

```diff
@@ -0,0 +1,14 @@
+gravitino.iceberg-rest.httpPort = 9001
+
+# --- Iceberg REST Catalog Backend (set to JDBC) ---
+gravitino.iceberg-rest.catalog-backend = jdbc
```

Comment:
The memory mode does still not work?

Reply:
Tested in memory mode and confirmed it does work correctly now with the manual sync between the container and host warehouse directories.

That said, I’m not sure memory mode should be the preferred option here. One of the goals of this PR is to demonstrate how HMSRestCatalogClient integrates with an external RESTCatalog server like Gravitino. For that reason, I think it makes more sense to highlight jdbc mode in the examples/tests, since it better reflects real-world usage where the catalog state is persisted beyond the lifetime of a single container.